zio/flags/flags.go

```diff
@@ -0,0 +1,85 @@
+package flags
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"regexp"
+
+	"github.com/brimsec/zq/zio/ndjsonio"
+	"github.com/brimsec/zq/zio/options"
+	"github.com/brimsec/zq/zio/zngio"
+)
+
+// Reader has the union of all the flags accepted by the different
+// Reader implementations.
+type Reader struct {
```

Comment:
Nit: As in zio/options, calling these types Reader and Writer feels off. I'd call them ReaderFlags and WriterFlags.

Reply:
Agreed.  See issue #1253